Fix broken modrun

### Problem

When attempting to start hedera-services using `modrun` gradle task, it appears to hang. On inspecting `swirlds.log`, the following error is present:

```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "com.hedera.pbj.runtime.io.buffer.Bytes.length()" because "item" is null
	at com.swirlds.platform.SwirldsPlatform.lambda$initializeState$7(SwirldsPlatform.java:736) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.threading.interrupt.Uninterruptable.abortAndThrowIfInterrupted(Uninterruptable.java:196) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.SwirldsPlatform.initializeState(SwirldsPlatform.java:729) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.SwirldsPlatform.<init>(SwirldsPlatform.java:437) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.Browser.createLocalPlatforms(Browser.java:708) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.Browser.startPlatforms(Browser.java:947) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.Browser.<init>(Browser.java:313) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.Browser.launch(Browser.java:541) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.Browser.parseCommandLineArgsAndLaunch(Browser.java:490) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.Browser.main(Browser.java:971) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "com.hedera.pbj.runtime.io.buffer.Bytes.length()" because "item" is null
	at com.swirlds.common.merkle.hash.FutureMerkleHash.rethrowException(FutureMerkleHash.java:83) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.merkle.hash.FutureMerkleHash.get(FutureMerkleHash.java:93) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.merkle.hash.FutureMerkleHash.get(FutureMerkleHash.java:30) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.platform.SwirldsPlatform.lambda$initializeState$7(SwirldsPlatform.java:734) ~[swirlds-platform-core-0.41.0-adhoc.x2d52beef.jar:?]
	... 9 more
Caused by: java.lang.NullPointerException: Cannot invoke "com.hedera.pbj.runtime.io.buffer.Bytes.length()" because "item" is null
	at com.hedera.node.app.spi.state.codec.BytesCodec.write(BytesCodec.java:42) ~[app-spi-0.41.0-SNAPSHOT.jar:?]
	at com.hedera.node.app.spi.state.codec.BytesCodec.write(BytesCodec.java:26) ~[app-spi-0.41.0-SNAPSHOT.jar:?]
	at com.hedera.node.app.state.merkle.singleton.ValueLeaf.serialize(ValueLeaf.java:106) ~[?:?]
	at com.swirlds.common.io.streams.SerializableDataOutputStream.writeSerializable(SerializableDataOutputStream.java:82) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.io.streams.SerializableDataOutputStream.writeSerializable(SerializableDataOutputStream.java:98) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.crypto.engine.SerializationDigestProvider.handleItem(SerializationDigestProvider.java:56) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.crypto.engine.SerializationDigestProvider.handleItem(SerializationDigestProvider.java:34) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.crypto.engine.OperationProvider.compute(OperationProvider.java:64) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.crypto.engine.OperationProvider.compute(OperationProvider.java:79) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.crypto.engine.CryptoEngine.digestSync(CryptoEngine.java:414) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.crypto.Cryptography.digestSync(Cryptography.java:171) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.merkle.crypto.internal.MerkleCryptoEngine.digestSync(MerkleCryptoEngine.java:124) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.merkle.hash.MerkleHashBuilder.hashSubtree(MerkleHashBuilder.java:211) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at com.swirlds.common.merkle.hash.MerkleHashBuilder.lambda$createHashingRunnable$1(MerkleHashBuilder.java:173) ~[swirlds-common-0.41.0-adhoc.x2d52beef.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]

```

### Solution

Fix it so that services can start again.

### Alternatives

_No response_

--- a/hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/FreezeServiceImpl.java
+++ b/hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/FreezeServiceImpl.java
@@ -23,6 +23,7 @@ import com.hedera.node.app.spi.state.Schema;
 import com.hedera.node.app.spi.state.SchemaRegistry;
 import com.hedera.node.app.spi.state.StateDefinition;
 import com.hedera.node.app.spi.state.codec.BytesCodec;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.Set;
 
@@ -47,7 +48,11 @@ public final class FreezeServiceImpl implements FreezeService {
 
             @Override
             public void migrate(@NonNull final MigrationContext ctx) {
-                // nothing to do
+                // Reset the upgrade file hash to empty
+                // It should always be empty at genesis or after an upgrade, to indicate that no upgrade is in progress
+                // Nothing in state can ever be null, so use Bytes.EMPTY to indicate an empty hash
+                final var upgradeFileHashKeyState = ctx.newStates().getSingleton(UPGRADE_FILE_HASH_KEY);
+                upgradeFileHashKeyState.put(Bytes.EMPTY);
             }
         };
     }

--- a/hedera-node/hedera-network-admin-service-impl/src/test/java/com/hedera/node/app/service/networkadmin/impl/test/FreezeServiceImplTest.java
+++ b/hedera-node/hedera-network-admin-service-impl/src/test/java/com/hedera/node/app/service/networkadmin/impl/test/FreezeServiceImplTest.java
@@ -18,8 +18,11 @@ package com.hedera.node.app.service.networkadmin.impl.test;
 
 import static com.hedera.node.app.service.networkadmin.impl.FreezeServiceImpl.UPGRADE_FILE_HASH_KEY;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.mockito.Mockito.verify;
 
+import com.hedera.node.app.fixtures.state.FakeHederaState;
+import com.hedera.node.app.fixtures.state.FakeSchemaRegistry;
 import com.hedera.node.app.service.networkadmin.FreezeService;
 import com.hedera.node.app.service.networkadmin.impl.FreezeServiceImpl;
 import com.hedera.node.app.spi.state.Schema;
@@ -43,7 +46,7 @@ class FreezeServiceImplTest {
         final FreezeService service = new FreezeServiceImpl();
 
         // then
-        Assertions.assertNotNull(service, "We must always receive an instance");
+        assertNotNull(service, "We must always receive an instance");
         Assertions.assertEquals(
                 FreezeServiceImpl.class,
                 service.getClass(),
@@ -64,5 +67,18 @@ class FreezeServiceImplTest {
         final var iter =
                 statesToCreate.stream().map(StateDefinition::stateKey).sorted().iterator();
         assertEquals(UPGRADE_FILE_HASH_KEY, iter.next());
+    }
+
+    @Test
+    void migratesAsExpected() {
+        final var subject = new FreezeServiceImpl();
+        final var registry = new FakeSchemaRegistry();
+        final var state = new FakeHederaState();
+
+        subject.registerSchemas(registry);
+        registry.migrate(FreezeService.NAME, state);
+        final var upgradeFileHashKeyState =
+                state.createReadableStates(FreezeService.NAME).getSingleton(UPGRADE_FILE_HASH_KEY);
+        assertNotNull(upgradeFileHashKeyState.get());
     }
 }

